### ISSUE TITLE
SC2062 / misleading comment

### ISSUE THREAD

**Issue by bittorf** [koalaman/shellcheck#517]
``` shell
#!/bin/sh
grep -- '-foo' dir/*
```

message is:
Quote the grep pattern so the shell won't interpret it.

**Comment by Arthur2e5** [comment]
Seems more like a false positive.

**Comment by deryni** [comment]
This appears to be shellcheck not treating `--` as the end of options and so mis-parsing what `-foo` is doing there. But I'm not sure how safe that assumption on shellcheck's part would be.

That it gets this wrong with `grep -e '-foo' dir/*` is also understandable but perhaps a bit more importantly incorrect (as it is the more portable solution to this problem I believe). Though supporting that requires even more specialized knowledge than understanding `--` requires.

**Comment by bittorf** [comment]
while we are at it, shellsheck should also warn, when doing e.g.
```shell
grep "$pattern" "$file"
```
because when $pattern begins with '-' it breaks, the same with e.g.
```
logger "$0: bla"
```
when $0 = '-ash'

**Comment by koalaman** [comment]
This particular case plus `-e -foo bar*` and `--regex -foo bar*` has now been fixed.

`cmd "$foo"` for when `$foo` contains a leading dash is very much an issue, though a bit of a separate one.
